app/models/devolved_administration_availability.rb

```diff
@@ -4,4 +4,27 @@ class DevolvedAdministrationAvailability
   embedded_in :local_transaction_edition
   field :type, type: String, default: "local_authority_service"
   field :alternative_url, type: String
+
+  validates :type, inclusion: { in: %w[local_authority_service devolved_administration_service unavailable] }
+  validates :alternative_url, presence: true, if: -> { devolved_administration_service? }
+  validate :alternative_url_format
+
+  def alternative_url_format
+    return if alternative_url.blank?
+
+    begin
+      uri = URI.parse(alternative_url)
+    rescue URI::Error => e
+      Rails.logger.debug e.message.to_s
+      errors.add(:alternative_url, "must be a full URL, starting with https://")
+    end
+
+    unless uri.is_a?(URI::HTTPS)
+      errors.add(:alternative_url, "must be a full URL, starting with https://")
+    end
```

Comment:
You can probably simplify this to avoid setting the error twice

```suggestion
    uri = begin
            URI.parse(alternative_uri)
          rescue URI::Error
            nil
          end
         
    unless uri.is_a?(URI::HTTPS)
      errors.add(:alternative_url, "must be a full URL, starting with https://")
    end         
```

Reply:
👍